app/src/main/java/com/example/habitsmasher/HabitDialog.java

```diff
@@ -0,0 +1,273 @@
+package com.example.habitsmasher;
+
+import android.app.DatePickerDialog;
+import android.content.Context;
+import android.util.Log;
+import android.widget.Button;
+import android.widget.DatePicker;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.DialogFragment;
+import com.example.habitsmasher.ui.dashboard.HabitListFragment;
+
+
+/**
+ * Abstract UI class that describes any dialog that can
+ * add or edit habits
+ * @author Jason Kim
+ */
+public abstract class HabitDialog extends DialogFragment implements DisplaysErrorMessages {
```

Comment:
This is the HabitDialog class, which is an abstract class in which EditHabitDialog and AddHabitDialog extend from.